src/js/fluxcontroller.js

```diff
@@ -272,13 +273,26 @@ define(function (require, exports, module) {
 
                 actionDependencies.set(action, dependencies);
 
-                var reads = action.reads || locks.ALL_LOCKS,
-                    writes = action.writes || locks.ALL_LOCKS;
+                var reads,
+                    writes;
+
+                if (actionObject) {
+                    reads = actionObject.reads || locks.ALL_LOCKS;
+                    writes = actionObject.writes || locks.ALL_LOCKS;
+                } else {
```

Comment:
we need to add some additional filtering if we want all of the "actions" that reach here to have an action property. I'll look into adding it.